framework/lib/components/select/get-matching-value.ts

```diff
@@ -0,0 +1,20 @@
+import { filter, find, is, length, propEq } from 'ramda'
+import { Option, SelectProps } from './types'
+
+export const getMatchingValue = <T extends Option>(
+  value: SelectProps<T>['value'],
+  options: SelectProps<T>['options']
+): T | T[] => {
+  if (typeof value === 'string') {
+    return find(propEq('value', value), options as T[]) as T
```

Comment:
to not cast so many things i think we could use type guards for this. 
eg.
`export const isIncludeFilter = (filter: IncludeFilter | ExcludeFilter): filter is IncludeFilter => '$in' in filter` cause when doing if on this the later lines will know the type without needing to cast